onnx/test/shape_inference_test.py

```diff
@@ -3881,7 +3895,7 @@ def test_optional_sequence_has_element(self) -> None:
              make_node('OptionalHasElement', ['optional'], ['output'])],
             [])
         self._assert_inferred(graph, [sequence_val_info, optional_val_info,
-                                      make_tensor_value_info('output', TensorProto.BOOL, None)])  # type: ignore
+                                      make_tensor_value_info('output', TensorProto.BOOL, ())])  # type: ignore
```

Comment:
Originally these 3 tests (which involve Optional-related operators) have a tensor whose shape is empty (has the shape field without any content). We should use `()` to represent this case.

A follow-up question: are rank 0 (having a shape field without any content) and empty shape (not having a shape field) always indifferent? From my understanding, rank 0 means it doesn't have (or need) a shape; empty shape means unknown shape. @gramalingam am I correct? Thank you.

Reply:
Let me look at the tests. But "rank 0" means same as "shape is an empty list" same as "empty shape". But "unknown shape" is different and is represented by "no shape field" and it means that the "rank is unknown".

Reply:
Using shape `()` or `[]` for "rank 0" makes sense to me. I need to look at the examples to see what is expected in the example.